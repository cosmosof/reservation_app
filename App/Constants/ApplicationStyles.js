import {Platform} from 'react-native';
import Fonts from './Fonts';
import Colors from './Colors';

const ApplicationStyles = {
  navbarHeader: {
    color: Colors.darkMatBlue2,
    fontSize: 18,
    fontWeight: '600',
    fontFamily: Platform.OS == 'ios' ? Fonts.typeIOS.condensed : Fonts.typeAndroid.condensed,
  },
};

export default {
  ApplicationStyles,
};
