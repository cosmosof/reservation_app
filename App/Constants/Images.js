const images = {
  logo: require('../Images/Relate.png'),
};

export default images;
