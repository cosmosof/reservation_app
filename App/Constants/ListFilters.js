export default [
  {
    index: 0,
    name: 'guest name',
    value: 'name',
  },
  {
    index: 1,
    name: 'in guest names',
    value: 'name_contains',
  },
  {
    index: 3,
    name: 'name starts with',
    value: 'name_starts_with',
  },
  {
    index: 4,
    name: 'hotel name',
    value: 'hotelName',
  },
  {
    index: 5,
    name: 'in hotel names',
    value: 'hotelName_contains',
  },
  {
    index: 7,
    name: 'hotel names start with',
    value: 'hotelName_starts_with',
  },
  {
    index: 9,
    name: 'arrival date',
    value: 'arrivalDate',
  },
  {
    index: 10,
    name: 'departure date',
    value: 'departureDate',
  },
];
