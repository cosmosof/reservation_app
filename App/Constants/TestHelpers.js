const noop = () => {};
const sampleText = 'Sample text';
export default {
  noop,
  sampleText,
};
