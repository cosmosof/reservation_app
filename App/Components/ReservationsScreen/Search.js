// @flow

import * as React from 'react';
import {TextInput, View, Platform} from 'react-native';
import Icon from 'react-native-vector-icons/Ionicons';
import {Colors} from '../../Constants';

import styles from './Styles/SearchStyles';

type Props = {
  children: Function,
  styles: Object,
};

type State = {
  text: string,
};

export default class Search extends React.Component<Props, State> {
  constructor(props: Object) {
    super(props)
    this.state = {
      text: '',
    };
  }
  
  updateText = (text: string) => {
    this.setState({text});
  };
  render() {
    const {text} = this.state;
    return (
      <View style={styles.searchBarContainer}>
        <View style={styles.searchBar}>
          <Icon
            color={Colors.charcoal}
            name={Platform.OS == 'ios' ? 'ios-search' : 'md-search'}
            size={20}
          />
          <TextInput
            autoCorrect={false}
            onChangeText={(text) => this.updateText(text)}
            placeholder="search reservations..."
            style={styles.textInput}
            value={text}
          />
        </View>
        {this.props.children(text)}
      </View>
    );
  }
}
