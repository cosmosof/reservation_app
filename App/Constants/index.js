import Colors from './Colors';
import Fonts from './Fonts';
import Metrics from './Metrics';
import Images from './Images';
import ListFilters from './ListFilters';
import ApplicationStyles from './ApplicationStyles';
import TestHelpers from './TestHelpers';

export {Colors, Fonts, Images, Metrics, ListFilters, ApplicationStyles, TestHelpers};
