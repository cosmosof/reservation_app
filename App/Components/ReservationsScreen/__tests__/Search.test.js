import React from 'react';
import {shallow} from 'enzyme';
import Search from '../Search';
import {TestHelpers} from '../../../Constants';

const {sampleText} = TestHelpers;

describe('<Search>', () => {
  describe('Structure', () => {
    const renderFn = jest.fn().mockReturnValue(null);
    it('renders correctly', () => {
      const wrapper = shallow(<Search>{renderFn}</Search>);
      expect(wrapper).toMatchSnapshot();
    });
  });
  describe('Functionality', () => {
    it('Calls the render props function', () => {
      const renderFn = jest.fn().mockReturnValue(null);
      const wrapper = shallow(<Search>{renderFn}</Search>);
      expect(renderFn.mock.calls.length).toBe(1);
      expect(wrapper.state('text')).toBe('');
    });
    it('sets state correctly', () => {
      const renderFn = jest.fn().mockReturnValue(null);
      const wrapper = shallow(<Search text={sampleText}>{renderFn}</Search>);
      wrapper.setState({text: sampleText});
      expect(wrapper.state('text')).toEqual(sampleText);
    });
    it('onChangeText sets state correctly', () => {
      const renderFn = jest.fn().mockReturnValue(null);
      const wrapper = shallow(<Search text={sampleText}>{renderFn}</Search>);
      wrapper.instance().updateText(sampleText);
      expect(wrapper.state('text')).toBe(sampleText);
    });
  });
});
