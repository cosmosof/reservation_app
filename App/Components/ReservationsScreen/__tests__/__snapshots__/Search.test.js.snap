// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Search> Structure renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "white",
      "flex": 1,
      "flexDirection": "row",
      "justifyContent": "center",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#F4F4F4",
        "borderColor": "#CCCCCC",
        "borderRadius": 4,
        "borderWidth": 0.5,
        "flexDirection": "row",
        "height": 36,
        "justifyContent": "center",
        "marginRight": 10,
        "paddingLeft": 10,
      }
    }
  >
    <Icon
      allowFontScaling={false}
      color="#595959"
      name="ios-search"
      size={20}
    />
    <TextInput
      allowFontScaling={true}
      autoCorrect={false}
      onChangeText={[Function]}
      placeholder="search reservations..."
      style={
        Object {
          "color": "#2d2d2d",
          "fontFamily": "AvenirNext-Regular",
          "fontSize": 14,
          "height": 40,
          "padding": 6,
          "width": 220,
        }
      }
      underlineColorAndroid="transparent"
      value=""
    />
  </View>
</View>
`;
